src/notebooks_service.py

```diff
@@ -170,18 +170,32 @@ def launch_notebook(user, namespace, project, commit_sha, notebook=None):
         return redirect(notebook_url)
 
     # 1. launch using spawner that checks the access
+    payload = {
+        'branch': request.args.get('branch', 'master'),
+        'commit_sha': commit_sha,
+        'namespace': namespace,
+        'notebook': notebook,
+        'project': project,
+    }
+    if os.environ.get('GITLAB_REGISTRY_SECRET'):
+        payload['image_pull_secrets'] = payload.get('image_pull_secrets', [])
```

Comment:
`payload` here will never contain `image_pull_secrets` -- or is that something you expect in the future?

Reply:
I thought it would make the code robust to changes anywhere.

Reply:
👍 just making sure I'm not missing something obvious